include/git2/remote.h

```diff
@@ -41,6 +41,52 @@ GIT_EXTERN(int) git_remote_create(
 		const char *name,
 		const char *url);
 
+/**
+ * Remote creation options structure
+ *
+ * Initialize with `GIT_REMOTE_CREATE_OPTIONS_INIT`. Alternatively, you can
+ * use `git_remote_create_init_options`.
+ *
+ */
+typedef struct git_remote_create_options {
+	unsigned int version;
+
+	/** The repository that should own the remote.
+	 * Setting this to NULL results in an in-memory/anonymous remote.
+	 */
+	git_repository *repository;
+
+	/** The remote's name.
+	 * Setting this to NULL results in an in-memory/anonymous remote.
+	 */
+	const char *name;
+
+	/** The fetchspec the remote should use. */
+	const char *fetchspec;
+} git_remote_create_options;
+
+#define GIT_REMOTE_CREATE_OPTIONS_VERSION 1
+#define GIT_REMOTE_CREATE_OPTIONS_INIT {GIT_REMOTE_CREATE_OPTIONS_VERSION}
+
+GIT_EXTERN(int) git_remote_create_init_options(
+		git_remote_create_options *opts,
+		unsigned int version);
+
+/**
+ * Create a remote, with options.
+ *
+ * This function allows more fine-grained control over the remote creation.
+ *
+ * @param out the resulting remote
+ * @param url the remote's url
+ * @param opts the remote creation options
+ * @return 0 on success, WIP
+ */
+GIT_EXTERN(int) git_remote_create_ext(
```

Comment:
Bikeshedding: `create_ext` doesn't make it clear what the "extended" nature of this function is. Instead, I'd vote for `git_remote_create_with_opts`, which clearly states that it takes an `opts` structure.